torch/nn/modules/module.py

```diff
@@ -1144,10 +1144,24 @@ def to(self, *args, **kwargs):
                     "if a complex module does not work as expected.")
 
         def convert(t):
-            if convert_to_format is not None and t.dim() in (4, 5):
-                return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None,
-                            non_blocking, memory_format=convert_to_format)
-            return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
+            try:
+                if convert_to_format is not None and t.dim() in (4, 5):
+                    return t.to(
+                        device,
+                        dtype if t.is_floating_point() or t.is_complex() else None,
+                        non_blocking,
+                        memory_format=convert_to_format,
+                    )
+                return t.to(
+                    device,
+                    dtype if t.is_floating_point() or t.is_complex() else None,
+                    non_blocking,
+                )
+            except NotImplementedError as e:
+                if str(e) == "Cannot copy out of meta tensor; no data!":
+                    raise NotImplementedError(
+                        f"{e} Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to()."
+                    )
```

Comment:
nit: From the [traceback you shared](https://github.com/pytorch/pytorch/pull/119353#discussion_r1482044339), seems like "Cannot copy out of meta tensor; no data!" occurs twice maybe

```suggestion
                    raise NotImplementedError(
                        "Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to()."
                    ) from e
```

Reply:
```
                    raise NotImplementedError(
                        f"{e} Please use torch.nn.Module.to_empty() instead of torch.nn.Module.to()."
                    ) from None
```
Yeah, amended to this and the error message only occur once, thanks for the suggestion.

Reply:
think it should be `from e`